local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,254 @@
+        return collision_zone_poly.buffer(COLLISION_ZONE_SAFETY_BUFFER, join_style=2)
+
+
+class Boat(Obstacle):
+    """Describes boat objects which Sailbot must avoid.
+    Also referred to target ships or boat obstacles.
+
+    Attributes:
+        id (int): MMSI ID of the boat.
+        width (float): Width of the boat in meters.
+        length (float): Length of the boat in meters.
+    """
+
+    def __init__(
+        self,
+        reference: LatLon,
+        sailbot_position: LatLon,
+        sailbot_speed: float,
+        ais_ship: HelperAISShip,
+    ):
+        super().__init__(reference, sailbot_position, sailbot_speed)
+
+        self.ais_ship = ais_ship
+        self.update_collision_zone(self.create_boat_collision_zone(self.ais_ship))
+
+    def create_boat_collision_zone(self, ais_ship: HelperAISShip) -> Polygon:
+        """Creates a Shapely Polygon that represents the boat's collision zone,
+        which is shaped like a cone.
+
+        Args:
+            ais_ship (HelperAISShip): AIS Ship message containing information about the boat.
+
+        Returns:
+            Polygon: Shapely Polygon representing the boat's collision zone.
+        """
+        if self.ais_ship.id != ais_ship.id:
+            raise ValueError("Argument AIS Ship ID does not match this Boat instance's ID")
+
+        # Ensure the instance variable ais_ship is the most up-to-date version
+        # Because this function may be called outside of the constructor
+        self.ais_ship = ais_ship
+
+        # coordinates of the center of the boat
+        position = latlon_to_xy(
+            self.reference, LatLon(ais_ship.lat_lon.latitude, ais_ship.lat_lon.longitude)
+        )
+
+        width = meters_to_km(self.ais_ship.width.dimension)
+        length = meters_to_km(self.ais_ship.length.dimension)
+
+        # Course over ground of the boat
+        cog = ais_ship.cog.heading
+
+        # Calculate distance the boat will travel before soonest possible collision with Sailbot
+        projected_distance = self.calculate_projected_distance(ais_ship)
+
+        # TODO This feels too arbitrary, maybe will incorporate ROT at a later time
+        collision_zone_stretch = projected_distance * COLLISION_ZONE_STRETCH_FACTOR
+
+        # Points of the boat collision cone polygon before rotation and centred at the origin
+        collision_zone_poly = Polygon(
+            [
+                [-width / 2, -length / 2],
+                [-collision_zone_stretch * width, length / 2 + projected_distance],
+                [collision_zone_stretch * width, length / 2 + projected_distance],
+                [width / 2, -length / 2],
+            ]
+        )
+
+        collision_zone_poly = af.rotate(collision_zone_poly, -cog, origin=(0, 0))
+
+        return Obstacle.create_collision_zone(collision_zone_poly, position)
+
+    def calculate_projected_distance(self, ais_ship: HelperAISShip) -> float:
```

Comment:
Why not add implementation in `calculate_time_to_intersection` here? This function is pretty small, but ultimately utu just a suggestion